src/mito2/src/compaction.rs

```diff
@@ -299,20 +276,86 @@ impl CompactionScheduler {
             for waiter in waiters {
                 waiter.send(Ok(0));
             }
-            return None;
+            self.region_status.remove(&region_id);
+            return Ok(());
         };
 
-        let task = CompactionTaskImpl {
+        // If specified to run compaction remotely, we schedule the compaction job remotely.
+        // It will fall back to local compaction if there is no remote job scheduler.
+        if current_version.options.compaction.remote_compaction() {
```

Comment:
I think we can add a step to evaluate the compaction job to see if it's suitable to run locally or remotely. The default implementation can be noop. This allows us add more strategy in future to control the behaviour.

Reply:
I agree with that. The current implementation decides whether to use remote compaction based on the region options specified with the table creation. I think we can put more strategy in the future.